x-pack/plugins/observability_solution/apm/server/routes/services/get_service_agent.ts

```diff
@@ -103,7 +103,7 @@ export async function getServiceAgent({
     return {};
   }
 
-  const { agent, service, cloud } = normalizeFields(response.hits.hits[0].fields) as ServiceAgent;
+  const { agent, service, cloud } = serviceAgentName(response.hits.hits[0].fields) as ServiceAgent;
```

Comment:
In most places where I used `as` we can replace if with return type  in the mapper- so the `serviceAgentName` return type should be `ServiceAgent` so we can get rid of the casting.